// @divriots/jampack/dist/cmd-serve.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const es_dev_server_1 = require("es-dev-server");
const fs_1 = __importDefault(require("fs"));
const path_1 = __importDefault(require("path"));
function createSymlinkToNodeModules(config) {
    const rootNodeModules = path_1.default.join(config.rootDir, "node_modules");
    try {
        fs_1.default.unlinkSync(rootNodeModules);
    }
    catch (e) {
        // No problem
    }
    try {
        fs_1.default.symlinkSync(path_1.default.join(process.cwd(), config.modulesDir), rootNodeModules);
    }
    catch (e) {
        // Let's assume it's all good
        console.error("Error creating symlink to node_modules", e);
    }
    return rootNodeModules;
}
function serve(config) {
    // Set development mode
    process.env.NODE_ENV = "development";
    const rootNodeModules = createSymlinkToNodeModules(config);
    es_dev_server_1.startServer(es_dev_server_1.createConfig({
        port: config.servePort,
        rootDir: config.rootDir,
        logStartup: true,
        logErrorsToBrowser: true,
        watch: true,
        nodeResolve: true,
        moduleDirs: [rootNodeModules],
        preserveSymlinks: true,
    }));
}
exports.default = serve;
