// @divriots/jampack/dist/config.js
const default_options = {
    image: {
        embed_size: 1500,
        srcset_min_width: 640,
        compress: true,
        jpeg: {
            options: {
                quality: 75,
                mozjpeg: true,
            },
        },
        png: {
            options: {
                compressionLevel: 9,
            },
        },
        webp: {
            options_lossless: {
                effort: 4,
                quality: 77,
                mode: 'lossless',
            },
            options_lossly: {
                effort: 4,
                quality: 77,
                mode: 'lossly',
            },
        },
    },
};
const fast_options_override = {
    image: {
        embed_size: 0,
        srcset_min_width: 16000,
        compress: false,
        jpeg: {
            options: {
                mozjpeg: false,
            },
        },
        png: {
            options: {
                compressionLevel: 0,
            },
        },
        webp: {
            options_lossless: {
                effort: 0,
            },
            options_lossly: {
                effort: 0,
            },
        },
    },
};
export function fast() {
    Object.assign(default_options, fast_options_override);
}
export default default_options;
